### PR TITLE
docs: alternative Rust client

### PR DESCRIPTION
**Pull request by davidhewitt**
To complement the "raw" Python example.

### CODE REVIEW

File: docs/guides/advanced/alternative-clients.md [hunk 1]
```diff
@@ -44,3 +44,70 @@ exporter = OTLPSpanExporter(
...
+
+```rust
+// src/main.rs
+use opentelemetry::{
+    global::ObjectSafeSpan,
+    trace::{Tracer, TracerProvider},
+};
+
+fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    // If you don't want to export environment variables, you could also just
```
Comment (alexmojaki) [inline docs/guides/advanced/alternative-clients.md]: Could this configuration be done in the exporter construction like in the Python example?

File: docs/guides/advanced/alternative-clients.md [hunk 2]
```diff
@@ -44,3 +44,70 @@ exporter = OTLPSpanExporter(
...
+
+```toml
+# Cargo.toml
+[package]
+name = "otel-example"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+opentelemetry = { version = "0.24", default-features = false, features = ["trace"] }
```
Comment (alexmojaki) [inline docs/guides/advanced/alternative-clients.md]: Does the version need to be specified? The current latest is 0.26. Same for the below.

File: docs/guides/advanced/alternative-clients.md [hunk 3]
```diff
@@ -44,3 +44,70 @@ exporter = OTLPSpanExporter(
...
+```sh
+cargo new --bin otel-example && cd otel-example
+export OTEL_EXPORTER_OTLP_ENDPOINT=https://logfire-api.pydantic.dev
+export OTEL_EXPORTER_OTLP_HEADERS='Authorization=your-write-token'
+```
+
+Update the `Cargo.toml` and `main.rs` files with the following contents:
+
+```toml
+# Cargo.toml
```
Comment (alexmojaki) [inline docs/guides/advanced/alternative-clients.md]:
```suggestion
```toml title="Cargo.toml"
```

File: docs/guides/advanced/alternative-clients.md [hunk 4]
```diff
@@ -44,3 +44,70 @@ exporter = OTLPSpanExporter(
...
 )
 ```
+
+
+## Example with Rust
+
+First, set up a new Cargo project:
+
+```sh
+cargo new --bin otel-example && cd otel-example
```
Comment (alexmojaki) [inline docs/guides/advanced/alternative-clients.md:54]: Is the command needed if you create `Cargo.toml` manually?

File: docs/guides/advanced/alternative-clients.md [hunk 5]
```diff
@@ -44,3 +44,70 @@ exporter = OTLPSpanExporter(
...
+    let tracer_provider = opentelemetry_otlp::new_pipeline()
+        .tracing()
+        .with_exporter(otlp_exporter)
+        .install_simple()?;
+    let tracer = tracer_provider.tracer("my_tracer");
+
+    tracer.span_builder("Hello World").start(&tracer).end();
+
+    tracer_provider.force_flush();
+    tracer_provider.shutdown()?;
```
Comment (alexmojaki) [inline docs/guides/advanced/alternative-clients.md]: Does it not automatically export on shutdown? Are both of these lines needed?

Comment (davidhewitt) [reply docs/guides/advanced/alternative-clients.md]: If the async client is used, these are probably a good idea. For blocking client, they're not necessary, will remove for now.

File: docs/guides/advanced/alternative-clients.md (hunk 4 above)
Comment (davidhewitt) [reply docs/guides/advanced/alternative-clients.md:54]: No, but it doesn't hurt to help align them with the file structure?

(For what it's worth, maybe in a year or two there will be single-file Rust scripts like https://rust-lang.github.io/rfcs/3424-cargo-script.html)

File: docs/guides/advanced/alternative-clients.md (hunk 2 above)
Comment (davidhewitt) [reply docs/guides/advanced/alternative-clients.md]: Updated to use `"*"`. I think unless we check this in CI it bitrots, but hey, we can live with that.

File: docs/guides/advanced/alternative-clients.md (hunk 1 above)
Comment (davidhewitt) [reply docs/guides/advanced/alternative-clients.md]: Yes, done.

File: docs/guides/advanced/alternative-clients.md [hunk 6]
```diff
@@ -44,3 +44,66 @@ exporter = OTLPSpanExporter(
...
+        .with_protocol(opentelemetry_otlp::Protocol::HttpBinary)
+        // If you don't want to export environment variables, you can also configure
+        // programmatically like so:
+        //
+        // .with_endpoint("https://logfire-api.pydantic.dev/v1/traces")
+        // .with_headers({
+        //     let mut headers = std::collections::HashMap::new();
+        //     headers.insert(
+        //         "Authorization".into(),
+        //         "NGCv6sl4PV1fC0W4BZS0dYFFGF7L9trDrf6MMnjJsX9x".into(),
```
Comment (alexmojaki) [inline docs/guides/advanced/alternative-clients.md]: .

Comment (davidhewitt) [reply docs/guides/advanced/alternative-clients.md]: Thanks and revoked it 😂

**Comment by alexmojaki** [comment]
Thanks!

**Pull request merged**